### PR TITLE
OBF Integration as discussed in #454

### PR DESCRIPTION
**Pull request by zelmon64**
This enhances the Open Board Format integration by using more of the information from the OBF files and making fewer assumptions. The discussion for this is in issue #454 but here are some highlights:

- Individual `.OBF` files can now be selected for use as well as `.OBZ` archives.
- The grid layout and key placement are now set by the information in the "grid" section of the OBF file.
- Support for inline and remote images.
- Support for local, inline and remote sounds.
- The defined button borders are now applied.

### CODE REVIEW

**Comment by JuliusSweetland** [comment]
@zelmon64 Thanks William. I will check this asap. Probably by the end of the weekend. I hope that's ok.

**Comment by zelmon64** [comment]
@JuliusSweetland yes that's fine with me thanks.

File: src/JuliusSweetland.OptiKey/Resources/Icons/SVG Source Files/CommuniKateToppage.svg [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (JuliusSweetland) [inline src/JuliusSweetland.OptiKey/Resources/Icons/SVG Source Files/CommuniKateToppage.svg]: Could you please change the casing of the filename (and any code uses of "Toppage") to "TopPage"

File: src/JuliusSweetland.OptiKey/UI/ValueConverters/IsActionKey.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Globalization;
+using System.Windows.Data;
+using JuliusSweetland.OptiKey.Models;
+
+namespace JuliusSweetland.OptiKey.UI.ValueConverters
+{
+    public class IsActionKey : IValueConverter
```
Comment (JuliusSweetland) [inline src/JuliusSweetland.OptiKey/UI/ValueConverters/IsActionKey.cs]: Could these new Key specific converters be renamed to include a reference to the fact that the key is a CommuniKate key

File: src/JuliusSweetland.OptiKey/UI/ValueConverters/IsSvgImage.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+using System.IO;
+
+namespace JuliusSweetland.OptiKey.UI.ValueConverters
+{
+    public class IsSvgImage : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            string image = value.ToString();
+            if (String.IsNullOrEmpty(image))
```
Comment (JuliusSweetland) [inline src/JuliusSweetland.OptiKey/UI/ValueConverters/IsSvgImage.cs]: string.

File: src/JuliusSweetland.OptiKey/UI/ValueConverters/IsValidImage.cs [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+using Microsoft.Win32;
+
+namespace JuliusSweetland.OptiKey.UI.ValueConverters
+{
+    public class IsValidImage : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            string image = value.ToString();
+            if (String.IsNullOrEmpty(image))
```
Comment (JuliusSweetland) [inline src/JuliusSweetland.OptiKey/UI/ValueConverters/IsValidImage.cs]: string. (I prefer using the keyword over the class - just a personal preference)

File: src/JuliusSweetland.OptiKey/UI/ValueConverters/IsValidImage.cs [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+        }
+    }
+
+    /// <summary>
+    /// Provides methods for checking whether a file can likely be opened as a BitmapImage, based upon its file extension
+    /// Source: http://james-ramsden.com/get-file-types-supported-by-bitmapimage/
+    /// </summary>
+    public class BitmapImageCheck : IDisposable
+    {
+        #region class variables
```
Comment (JuliusSweetland) [inline src/JuliusSweetland.OptiKey/UI/ValueConverters/IsValidImage.cs]: Could you please Title Case regions? Personal preference again.

**Comment by JuliusSweetland** [comment]
@zelmon64 You probably received an automatic notification, but if not I've added a handful of very minor review comments

**Comment by zelmon64** [comment]
@JuliusSweetland I found some other instances of `String.IsNullOrEmpty` and changed them to `string.IsNullOrEmpty` as well.

**Comment by JuliusSweetland** [comment]
@zelmon64 Merged - thank you William

**Pull request merged**